### DISCUSSION TITLE
Best method to detect when a placed offer is partially and/or fully filled?

### DISCUSSION THREAD

**Discussion by ZeroShaun** [XRPLF/xrpl.js#2529] [Q&A] [answered]
I'm having trouble figuring out how to get alerts when an order that was placed was partially or fully filled in the order books. I tried subscribing to the account that placed the order, but the transaction that triggers the event listener seems to only return data about the created offer which filled the order from the account I'm watching, without specific details about the actual amount(s) filled.

**Comment by mvadari** [comment]
There isn't a way to directly do this. I'd recommend subscribing to the `ledger` stream and just querying the object via `ledger_entry` every ledger.

**Reply by ZeroShaun** [reply]
Is the particular data I'm interested in even in the ledger stream data? What exactly would I need to look for (the ledger return object can be quite complex)?

**Reply by mvadari** [reply, marked as answer]
The ledger stream data is just useful for telling you when a ledger has closed, so you can requery the ledger object. Here's some code (I didn't actually run this, just an example):
```js
client.on("ledgerClosed", (ledger) => {
  const data = await client.request({command: "ledger_entry", index: "<fill in the offer's ID>"})
  // if the object doesn't exist then it was deleted and you can take a closer look at the transactions
  // if the object does exist, take a look at the `PreviousTxnID` and `PreviousTxnLgrSeq` values to see what transaction/ledger last modified it
})
```
